pkg/distributor/distributor.go

```diff
@@ -33,6 +39,10 @@ import (
 )
 
 const (
+	// ShardLbName is the internal label to be used by Loki when dividing a stream into smaller pieces.
+	// Possible values are only increasing integers starting from 0.
+	ShardLbName = "__stream_shard__"
```

Comment:
Aren't we also using `__shard` for splitting queries? I wonder if this could be confusing.

Reply:
it has the `__stream` prefix so IMO it is distinguishing well. Do you have a suggestion? We could try not using the `shard` word but I think it would be more confusing than helpful, as we're clearly sharding things.